src/utils.h

```diff
@@ -4,6 +4,10 @@
 
 #include <glib.h>
 
+#define STR_EMPTY(s) (*s == '\0')
+#define STR_NULL_OR_EMPTY(s) (!s || STR_EMPTY(s))
```

Comment:
At this point I'm not sure about the usefulness of this macro, it's longer than typing out `(!s || *s == '\0')` directly.

Reply:
Well yeah, that's true. But it's definitely easier to read and understand.

Do you have a better solution to cope GCC's weird understanding of NULL-pointer guards? https://travis-ci.org/dunst-project/dunst/jobs/437820396#L730-L735

Reply:
To be fair, there's nothing weird about it, `buf` in that warning is on the stack so it will always exist.

I don't think there's a better solution.

Reply:
I wasn't satisfied with this either.

We'll keep only `STR_EMPTY` and `STR_FULL` and throw `STR_NULL_OR_EMPTY` away.

I got the insight, that most standard library methods work only on a char buffer and not just do something with the buffer, but also propagate the length of the buffer they used.

So testing your stack buffer via `STR_EMPTY` is actually pointless, because it's a char buffer and not a string.